src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java

```diff
@@ -384,67 +418,74 @@ public void simulationPeriodic() {
 
     switch (state) {
       case IDLE:
-        sensors.setSimState(CoralEnum.NO_CORAL);
+        RobotContainer.sensors.setSimState(CoralEnum.NO_CORAL);
 
         if (atSetpoint(kPivotExtendAngle) && intakeMotor.get() > 0) {
-          t.start();
+          t1.start();
         } else {
-          t.stop();
+          t1.stop();
         }
 
-        if (t.hasElapsed(0.5)) {
+        if (t1.hasElapsed(0.5)) {
           state = IntakeState.FLOOR_INTAKE;
         }
 
         if (atSetpoint(kcoralStation) && intakeMotor.get() > 0) {
-          t.start();
+          t2.start();
         } else {
-          t.stop();
+          t2.stop();
         }
 
-        if (t.hasElapsed(0.5)) {
+        if (t2.hasElapsed(0.5)) {
           state = IntakeState.HP_CORAL_INTAKE;
         }
 
         if (atSetpoint(kalgae) && intakeMotor.get() < 0) {
-          t.start();
+          t3.start();
         } else {
-          t.stop();
+          t3.stop();
         }
 
-        if (t.hasElapsed(0.5)) {
+        if (t3.hasElapsed(0.5)) {
           state = IntakeState.ALGAE_INTAKE;
         }
       case FLOOR_INTAKE:
-        sensors.setSimState(CoralEnum.CORAL_NOT_ALIGNED);
+        RobotContainer.sensors.setSimState(
+            Math.random() > 0.5 ? CoralEnum.CORAL_TOO_CLOSE : CoralEnum.CORAL_TOO_FAR);
         state = IntakeState.LOCATE_CORAL;
+        t1.reset();
       case FLOOR_OUTTAKE:
         break;
       case HP_CORAL_INTAKE:
-        sensors.setSimState(CoralEnum.CORAL_NOT_ALIGNED);
+        RobotContainer.sensors.setSimState(
+            Math.random() > 0.5 ? CoralEnum.CORAL_TOO_CLOSE : CoralEnum.CORAL_TOO_FAR);
         state = IntakeState.LOCATE_CORAL;
+        t2.reset();
       case ALGAE_INTAKE:
+        Commands.waitSeconds(0.5);
         state = IntakeState.ALGAE_HOLD;
       case ALGAE_HOLD:
+        Commands.waitSeconds(0.5);
         state = IntakeState.ALGAE_OUTTAKE;
       case ALGAE_OUTTAKE:
+        Commands.waitSeconds(0.5);
         if (intakeMotor.get() > 0) {
           state = IntakeState.IDLE;
         }
       case LOCATE_CORAL:
-        if (atSetpoint(kPivotRetractAngle) && intakeMotor.get() > 0 && conveyorMotor.get() > 0) {
-          t.start();
+        if (atSetpoint(kcoralStation) && intakeMotor.get() > 0 && conveyorMotor.get() > 0) {
+          t4.start();
         } else {
-          t.stop();
+          t4.stop();
         }
 
-        if (t.hasElapsed(0.5)) {
-          sensors.setSimState(CoralEnum.CORAL_ALIGNED);
+        if (t4.hasElapsed(0.5)) {
```

Comment:
You should reset the timers inside these hasElapsed if blocks so the simulation can repeat the actions